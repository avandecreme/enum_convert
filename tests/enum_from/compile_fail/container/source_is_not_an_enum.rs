use enum_convert::EnumFrom;

struct Source {
    Field: i32,
}

#[derive(EnumFrom)]
#[enum_from(Source)] // Source should be an enum
enum Target {
    #[enum_from]
    Field,
}

fn main() {}
