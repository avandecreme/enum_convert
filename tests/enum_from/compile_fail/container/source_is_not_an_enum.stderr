error[E0599]: no associated item named `Field` found for struct `Source` in the current scope
  --> tests/enum_from/compile_fail/container/source_is_not_an_enum.rs:11:5
   |
3  |   struct Source {
   |   ------------- associated item `Field` not found for this struct
...
8  |   #[enum_from(Source)] // Source should be an enum
   |  _____________-
9  | | enum Target {
10 | |     #[enum_from]
11 | |     Field,
   | |    -^^^^^ associated item not found in `Source`
   | |____|
   |
